soc/nordic/common/uicr/gen_uicr/Kconfig

```diff
@@ -32,6 +32,15 @@ config GEN_UICR_SECURESTORAGE
 	  - At least one subpartition must be defined
 	  - Combined subpartition sizes must equal secure_storage_partition size
 
+config GEN_UICR_LOCK
+	bool "Enable UICR.LOCK"
+	help
+	  When enabled, the UICR generator will lock the entire UICR
+	  configuration in NVR0. Once locked, the UICR can only be modified
```

Comment:
Doesn't it lock all NVR0? Should we specify that its the NVR0 in MRAM10? Which includes BICR.